DEVDOCS.md

```diff
@@ -0,0 +1,85 @@
+# Enzyme-JAX Developer Documentation
+
+## Overview
+
+Enzyme-JAX is a C++ project that integrates the Enzyme automatic differentiation tool with JAX, enabling automatic differentiation of external C++ code within JAX. The project uses LLVM's MLIR framework for intermediate representation and transformation of code.
+
+## Building the Project
+
+### Quick Build
+```bash
+bazel build --repo_env=CC=clang-18 --color=yes --copt=-fbracket-depth=1024 --host_copt=-fbracket-depth=1024 -c dbg :enzymexlamlir-opt
```

Comment:
in princpile the bracket depth shouldnt be required here [and is added via the .bazelrc]

Reply:
fwiw for me `--copt=-fbracket-depth=1024` is necessary, `--host_copt=-fbracket-depth=1024` is not, and also these are currently commented out in the .bazelrc 🙃 so they should probably be uncommented there or put back in here

Reply:
without it, I get:
```
bazel-out/k8-dbg/bin/external/llvm-project/mlir/include/mlir/Dialect/Transform/IR/TransformDialect.h.inc:110:46: note: use -fbracket-depth=N to increase maximum nesting level
  110 |       (addOperationIfNotRegistered<OpTys>(), ...);
```